client/src/containers/ProfileCreation/ProfileCreationMaster.js

```diff
@@ -0,0 +1,195 @@
+import React, { useState, useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import { motion } from 'framer-motion';
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  NavLink,
+} from 'react-router-dom';
+import ProfileCreation1 from './ProfileCreation1/index';
+import ProfileCreation2 from './ProfileCreation2';
+import FilledButton from '../../components/filledButton/index';
+import Navbar from '../../components/navbar/index';
+
+const ProfileCreationMaster = () => {
+  const [currentPath, setCurrentPath] = useState(window.location.pathname);
+  const clubNameInput = useRef(null);
+  const shortDescInput = useRef(null);
+  const longDescInput = useRef(null);
+  const [errorMessage, setErrorMesssage] = useState('');
+  const [clubProfile, setClubProfile] = useState({
+    tags: [],
+    clubName: '',
+    shortDesc: '',
+    longDesc: '',
+    size: '',
+    memberPeriod: [],
+    requireApplication: '',
+    meetTime: ['', ''],
+  });
+
+  useEffect(() => {
+    console.log(clubProfile);
+  }, [clubProfile]);
+
+  const nextPage1 = () => {
+    let tempProfile = { ...clubProfile };
+    tempProfile.clubName = clubNameInput.current.value;
+    tempProfile.shortDesc = shortDescInput.current.value;
+    tempProfile.longDesc = longDescInput.current.value;
+
+    // Left empty
+    if (
+      tempProfile.clubName === '' ||
+      tempProfile.shortDesc === '' ||
+      tempProfile.longDesc === '' ||
+      tempProfile.size === '' ||
+      tempProfile.memberPeriod === [] ||
+      tempProfile.requireApplication === '' ||
+      tempProfile.tags === []
+    ) {
+      setErrorMesssage('Mandatory Field Missing!');
+      return;
+    } else if (
+      tempProfile.shortDesc.length > 150 ||
+      tempProfile.longDesc.length > 500
+    ) {
+      setErrorMesssage('Character Limit Exceeded!');
+      return;
+    }
+    setClubProfile(tempProfile);
+    setErrorMesssage('');
+  };
+
+  // Updates state based on input text
+  return (
+    <>
+      <Navbar />
+      <PageContainer>
+        <SetUpClubProfile />
+        {currentPath === '/profile-creation/1' && (
+          <PageDesc>Get-to-Know-You's: Why Students Should Join</PageDesc>
+        )}
+        {currentPath === '/profile-creation' && (
+          <PageDesc>Opening Statement: Provide Basic Info</PageDesc>
+        )}
+        <Router>
```

Comment:
we may want to document how this works a bit-- could be confusing for people coming into the codebase

Reply:
Added a comment to add some clarity.